R/get_inboveg_layer_information.R

```diff
@@ -0,0 +1,123 @@
+#'
+#' # get layer qualifiers from several specific surveys
+#' qualifiers_severalsurveys <- get_inboveg_layerinfo(con, survey_name =
+#' c("MILKLIM_Heischraal2012", "NICHE Vlaanderen"), multiple = TRUE)
+#'
+#' # get all layer qualifiers of all surveys
+#' allqualifiers <- get_inboveg_layerinfo(con)
+#'
+#' # Close the connection when done
+#' dbDisconnect(con)
+#' rm(con)
+#' }
+#'
+
+
+get_inboveg_layerinfo <- function(connection,
+                                       survey_name,
+                                       multiple = FALSE) {
+
+  assert_that(inherits(connection, what = "Microsoft SQL Server"),
+              msg = "Not a connection object to database.")
+
+  if (missing(survey_name) & !multiple) {
+    survey_name <- "%"
+  }
+
+  if (missing(survey_name) & multiple) {
+    stop("Please provide one or more survey names to survey_name when multiple
+         = TRUE")
+  }
+
+  if (!missing(survey_name)) {
+    if (!multiple) {
+      assert_that(is.character(survey_name))
+    } else {
+      assert_that(is.vector(survey_name, mode = "character"))
+    }
+  }
+
+common_part <- "SELECT ivS.Name
+                        , ivRecording.RecordingGivid
+                        , ivRecording.UserReference
+                        , ivRLLayer.LayerCode
+                        , ftAGV.Description as LayerDescription
+                        , ivRLLayer.CoverCode
+                        , ftAGV_01.Description as Percentage
+              FROM ivRecording
+              INNER JOIN ivSurvey ivS on ivS.Id = ivRecording.SurveyId
+              INNER JOIN  ivRLLayer on ivRLLayer.RecordingID = ivRecording.Id
+                  INNER JOIN ivRLResources on ivRLResources.ResourceGIVID = ivRLLayer.LayerResource
+                      LEFT JOIN [syno].[Futon_dbo_ftActionGroupValues] ftAGV ON ivRLResources.ListName = ftAGV.ListName COLLATE Latin1_General_CI_AI
+                      AND ivRLResources.ActionGroup = ftAGV.ActionGroup COLLATE Latin1_General_CI_AI
+                      AND ivRLLayer.LayerCode = ftAGV.code COLLATE Latin1_General_CI_AI
+                  INNER JOIN ivRLResources ivRLR_01 on ivRLR_01.ResourceGIVID = ivRLLayer.CoverResource
+                      LEFT JOIN [syno].[Futon_dbo_ftActionGroupValues] ftAGV_01 ON ivRLR_01.ListName = ftAGV_01.ListName COLLATE Latin1_General_CI_AI
+                      AND ivRLR_01.ActionGroup = ftAGV_01.ActionGroup COLLATE Latin1_General_CI_AI
+                      AND ivRLLayer.CoverCode = ftAGV_01.code COLLATE Latin1_General_CI_AI
+              WHERE 1 = 1"
+
+  if (!multiple) {
+    sql_statement <- glue_sql(common_part,
+                              "AND ivS.Name LIKE {survey_name}",
+                              survey_name = survey_name,
+                              .con = connection)
+
+  } else {
+    sql_statement <- glue_sql(common_part,
+                              "AND ivS.Name IN {survey_name*}",
```

Comment:
Is the `*` after `survey_name` necessary? I think it should be deleted and it may be the reason why the example with multiple survey names failed

Reply:
Ignore that. The problem is missing brackets: it should be `({survey_name*})`